### PR TITLE
Update changelog for Themis 0.12.0

### PR DESCRIPTION
**Pull request by ilammy**
<details>
<summary>Can you keep a secret?</summary>
It seems we’re going to make a release soon.
</details>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,310 @@
...
+- **Core**
+
+  - **Soter** (low-level security core used by Themis)
+
+    - New function: `soter_wipe()`.
+      It can be used to securely wipe sensitive data from memory after it's no longer necessary.
+      All Themis cryptosystems now make use of this new API.
+      ([#488](https://github.com/cossacklabs/themis/pull/488))
+
+    - Non-cryptographically secure PRNG could be used if the backend does not provide a cryptographically secure one.
```
Comment (vixentael) [inline CHANGELOG.md]:
```suggestion
    - Improved usage and error handling of OpenSSL/BoringSSL PRNGs.
```

actually, i don't see how this commit solves "Non-cryptographically secure PRNG could be used if ..."

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,310 @@
...
+      Thanks to [**@vixentael**](https://github.com/vixentael) for finding this issue.
+      ([#500](https://github.com/cossacklabs/themis/pull/500))
+
+  - **Secure Cell**
+
+    - Improved handling of edge cases with large data buffers.
+      ([#496](https://github.com/cossacklabs/themis/pull/496),
+       [#497](https://github.com/cossacklabs/themis/pull/497))
+
+    - Minor performance improvements in encryption/decryption code path.
```
Comment (vixentael) [inline CHANGELOG.md]:
```suggestion
    - Improved performance of SecureCell's encryption/decryption procedures.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,310 @@
...
+      ([#496](https://github.com/cossacklabs/themis/pull/496))
+
+  - **Library ABI**
+
+    - Themis shared libraries now have a formal ABI version.
+      This ensures that no compatibility issues arise if we ever need to introduce breaking changes in ABI.
+      ([#454](https://github.com/cossacklabs/themis/pull/454))
+
+    - Remove private symbols from public export lists.
```
Comment (vixentael) [inline CHANGELOG.md]:
```suggestion
    - Removed private symbols from public export lists.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,310 @@
...
+      Please consider using [binary repositories](https://docs.cossacklabs.com/pages/documentation-themis/#installing-themis-from-repositories) to install Themis.
+      If your system or package manager is not supported yet, please let us know via dev@cossacklabs.com.
+
+    - Themis packages now support multiarch installations.
+      ([#512](https://github.com/cossacklabs/themis/pull/512))
+
+      Multiarch enables parallel installation of 32-bit and 64-bit versions of the library.
+      This is particularly important on CentOS where some tools like _pkg-config_ failed to locate Themis due to non-standard installation path.
+
+    - Makefile now supports a number of [standard GNU variables](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html) like `libdir`.
```
Comment (vixentael) [inline CHANGELOG.md]:
```suggestion
    - Updated Makefile to support a number of [standard GNU variables](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html) like `libdir`.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,310 @@
...
+      ([#453](https://github.com/cossacklabs/themis/pull/453),
+       [#455](https://github.com/cossacklabs/themis/pull/455))
+
+    - Accurate dependencies for development packages make sure you don't have to install anything manually after installing Themis from package repositories.
```
Comment (vixentael) [inline CHANGELOG.md]:
```suggestion
    - Made more accurate dependencies for development packages make sure you don't have to install anything manually after installing Themis from package repositories.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,310 @@
...
+      ([#446](https://github.com/cossacklabs/themis/pull/446))
+
+    - NSIS installer is now available for Windows.
```
Comment (vixentael) [inline CHANGELOG.md]: add smth about this in `TLDR` section? it might be a large change for some of our users

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,310 @@
...
+
+- **iOS, macOS**
+
+  - Example code and projects for Objective-C and Swift are now up-to-date, cleaned up, and modernized.
+    ([#463](https://github.com/cossacklabs/themis/pull/463),
+     [#467](https://github.com/cossacklabs/themis/pull/467))
+
+  - OpenSSL version is pinned to 1.0.2.17 for Carthage package to work around a compilation issue with the latest version.
+    ([#484](https://github.com/cossacklabs/themis/pull/484))
+    <!-- TODO: Isn't this scheduled for a fix in 0.12? -->
```
Comment (vixentael) [inline CHANGELOG.md]: it is, if new OpenSSL would work well.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,310 @@
...
+## [0.12.0](https://github.com/cossacklabs/themis/releases/tag/0.12.0), September 23rd 2019
+
+**TL;DR:**
+Added WasmThemis to support WebAssembly,
+improved packaging and platform compatibility,
+additional safety checks and tricky bug fixes.
+
+**Breaking changes:**
+
+- Default installation path has been changed from `/usr` to `/usr/local`.
```
[the remark this answers is not in the document]
Comment (ilammy) [reply CHANGELOG.md]: On macOS it has been installed to `/usr/local` since forever. I guess we can say ‘on Linux’ then.

File: CHANGELOG.md (hunk 1 above)
Comment (ilammy) [reply CHANGELOG.md]: `RAND_bytes()` could return negative value to indicate that CSPRNG is not available and leave the buffer in undefined state (no idea what OpenSSL actually does there). We were treating this error code as success and went on with our business, assuming that we got the random bytes we needed. With this change we correctly report an error and do not assume that ‘random crap found on the call stack’ is a PRNG in any capacity.

Anyway, I like your wording better for being concise and significant.

**Comment by ilammy** [comment]
Okay, this quickly becomes unwieldy. I thought GitHub’s interface has improved, but apparently it still is not well suited for direct edits.

I’ll merge this PR into `pre-release-0.12.0` branch in the main repo and we can continue editing and reviewing there.

**Pull request merged**